library/kani_macros/src/sysroot/loop_contracts/mod.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Implementation of the loop contracts code generation.
+//!
+
+use proc_macro::TokenStream;
+use proc_macro_error2::abort_call_site;
+use quote::{format_ident, quote};
+use syn::spanned::Spanned;
+use syn::{Expr, Stmt};
+
+fn generate_unique_id_from_span(stmt: &Stmt) -> String {
```

Comment:
nit: I recommend to move the private functions to the bottom of the file for readability sake.